src/Bitfinex.coffee

```diff
@@ -160,7 +163,8 @@ module.exports = class Bitfinex
 		new Promise (resolve, reject) ->
 			request requestObj, (err, response, body) ->
 				if err || response.statusCode != 200
-					reject(statusCode: response.statusCode, body: body, err: err)
+					statusCode = if response then response.statusCode else null
+					return reject(statusCode: statusCode, body: body, err: err)
```

Comment:
I think you should do the statusCode variable assignment before the if statement check for != 200?